chore: cut v0.79.0 (#7194)
prometheus-operator/prometheus-operator: CHANGELOG.md

# 0.79.0 / 2024-12-13

> [!IMPORTANT]

> > With this release operator will be deploying Prometheus `v3.0.1` as default version.

Reviewer: ```suggestion
> With this release, the operator will be deploying Prometheus `v3.0.1` unless a specific version is specified.
```

> > Users who need Prometheus v2 would need to set it explicitly using `version` field under

Reviewer: ```suggestion
> Users who want to stay with Prometheus v2 need to set the `version` field under
```

> `Prometheus/PrometheusAgent` Spec.

* [CHANGE/FEATURE] Set Prometheus `v3.0.1` as default version installed by operator. #7158

> * [CHANGE] Add more API validations and refactoring in ScrapeConfig CRD. #6738 #7006

Reviewer: ```suggestion
* [CHANGE] Add more API validations to the ScrapeConfig CRD. #6738 #7006
```

* [FEATURE] Set Alertmanager API version to v2 for Prometheus v3 and above. #7102

> * [FEATURE] `scrape_classic_histograms` renamed to `always_scrape_classic_histograms` in Prometheus v3. #7104

Reviewer: Shouldn't be mentioned in the release notes?

> * [FEATURE] Support `--web.enable-otlp-receiver` flag in the Prometheus/PrometheusAgent CRDs. #7105

Reviewer: same here

* [FEATURE] Add support for `TranslationStrategy` option for OTLP receiver. #7112

> * [FEATURE] Support `metric_name_validation_scheme` option in Prometheus global configuration. #7116

Reviewer: we need to use the CRD field name rather than the Prometheus config field.

> * [FEATURE] Support `fallback_scrape_protocol` option in `Prometheus`, `PrometheusAgent`, `ServiceMonitor`, `PodMonitor`, `Probe`, `ScrapeConfig` and globally. #7131

Reviewer: same here

* [FEATURE] Add support for labels in PrometheusRule CRD at rules level. #7140
* [FEATURE] Add support for ProxyConfig in Alertmanager Endpoints. #7191
* [FEATURE] Add support for configuring Alertmanager STS `persistentVolumeClaimRetentionPolicy`. #7183
* [FEATURE] Add `enableHTTP2` to ScrapeConfig CRD. #7024
* [FEATURE] Add `filter` field in `ConsulSDConfig` in ScrapeConfig CRD #7124
* [FEATURE] Add `messageThreadID` field to TelegramConfig in AlertmanagerConfig CRD. #6999
* [ENHANCEMENT] Allow users to control the sharding label(s). #7126

> * [ENHANCEMENT] Add support for role selectors on PodMonitor. #7185

Reviewer: can we mention the new field name?
